### PR TITLE
Centralize credentials code in API

### PR DESCRIPTION
**Pull request by tamirkamara**
## What is being addressed

We had multiple places where we got credentials for the API actions on Azure which made it harder to make changes.

## How is this addressed

- Write the logic once and reuse across the application
- If managed identity is present, disable all other forms of getting a token in _DefaultAzureCredentials_ to remove some noise from the logs.
- Remove azure-mgmt-storage and update version of a couple of other packages
- Fix some pytest warnings

### CODE REVIEW

File: api_app/core/credentials.py [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+from azure.identity.aio import (
+    DefaultAzureCredential as DefaultAzureCredentialASync,
+    ManagedIdentityCredential as ManagedIdentityCredentialASync,
+    ChainedTokenCredential as ChainedTokenCredentialASync,
+)
+
+
+def get_credential() -> TokenCredential:
+    managed_identity = config.MANAGED_IDENTITY_CLIENT_ID
+    if managed_identity:
```
Comment (yuvalyaron) [inline api_app/core/credentials.py:19]: maybe it's worth logging whether we use the managed identity or not

**Review by yuvalyaron — CHANGES_REQUESTED** [review]

File: api_app/core/credentials.py (hunk 1 above)
Comment (tamirkamara) [reply api_app/core/credentials.py:19]: Is it?
On Azure it will always be managed, so this is about a local run.

**Review by yuvalyaron — APPROVED** [review]

**Pull request merged**